krux/parser.py

```diff
@@ -0,0 +1,318 @@
+                               uses the title of the _ArgumentGroup that this is wrapping. If set to False,
+                               does not add any prefix.
+        :type env_var_prefix: str | bool
+        :param args: Ordered arguments passed directly to argparse._ArgumentGroup.__init__()
+        :type args: list
+        :param kwargs: Keyword arguments passed directly to argparse._ArgumentGroup.__init__()
+        :type kwargs: dict
+        :return: The created KruxGroup object
+        :rtype: krux.parser.KruxGroup
+        """
+        group = KruxGroup(env_var_prefix=env_var_prefix, container=self, *args, **kwargs)
+        self._action_groups.append(group)
+        return group
+
+
+class KruxGroup(_ArgumentGroup):
+    HELP_ENV_VAR = "(env: {key})"
+    HELP_DEFAULT = "(default: {default})"
+
+    def __init__(self, env_var_prefix=None, *args, **kwargs):
+        """
+        Creates a wrapper around argparse._ArgumentGroup that handles some help doc automation as well as environment
+        variable fall back
+
+        :param env_var_prefix: Prefix to use for the environment variables. If set to None, uses the title
+                               of the _ArgumentGroup that this is wrapping. If set to False, does not add any prefix.
+        :type env_var_prefix: str | bool
+        :param args: Ordered arguments passed directly to krux.wrapper.Wrapper.__init__()
+        :type args: list
+        :param kwargs: Keyword arguments passed directly to krux.wrapper.Wrapper.__init__()
+        :type kwargs: dict
+        """
+        # Call to the superclass to bootstrap.
+        super(KruxGroup, self).__init__(*args, **kwargs)
+
+        if env_var_prefix is None:
+            self._env_prefix = self.title + '_'
+        elif env_var_prefix is False:
+            self._env_prefix = ''
+        else:
+            self._env_prefix = str(env_var_prefix) + '_'
+
+    def add_argument(self, *args, **kwargs):
+        """
+        Creates a CLI argument under this group based on the passed parameters.
+
+        :param env_var: Name of the environment variable to use. If set to None or not set, uses the name of
+                        the option and the env_var_prefix passed to the group to deduce a name. If set to False,
+                        does not add any environment variable support.
+        :type env_var: str | bool
+        :param add_env_var_help: Whether to add the name of the environment variable to the help text. If set to False
+                                 or if env_var is set to False, the help text is not appended.
+        :type add_env_var_help: bool
+        :param add_default_help: Whether to add the default value of the argument to the help text. If set to False
+                                 or if the help text already includes it, it is not appended.
+        :type add_default_help: bool
```

Comment:
There are actions like `store_true` where displaying the default value is little clumsy. Instead of me trying to figure out every possible use case, I just added a flag for it. `add_env_var_help` is there mostly for consistency.

This is somewhat poorly documented because `argparse` itself relies on `*args, **kwargs` notation, but both `add_env_var_help` and `add_default_help` are optional. If you don't set anything, I assume them to be `True`.